### PR TITLE
#16: Create lbaf-viz interface 

### PR DESCRIPTION
**Pull request by pierrepebay**
Fixes: #16

### CODE REVIEW

**Comment by cwschilly** [comment]
We're going to pause the work on CI for now and pick it up in a separate PR

File: bindings/python/tv.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#include <nanobind/stl/vector.h>
+
+#include <filesystem>
+#include <map>
+
+#include <omp.h>
+
+namespace vt::tv::bindings::python {
+
+void tv_from_json(const std::vector<std::string>&, const std::string&, uint64_t);
```
Comment (lifflander) [inline bindings/python/tv.h:72]: Change this to camel case to match style guidelines.

File: ci/build-and-test-bindings.dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM pierrpebay/vt-tv:master AS build
```
Comment (lifflander) [inline ci/build-and-test-bindings.dockerfile:1]: We should probably shouldn't use your account here. Do you have access to push to the main one?

File: src/vt-tv/render/render.cc [hunk 3]
```diff
@@ -76,6 +76,16 @@ Render::Render(Info in_info)
   }
   max_o_per_dim_ = 0;
 
+  // Normalize communication edges
+  for(PhaseType phase = 0; phase < this->n_phases_; phase++) {
+    if (
```
Comment (lifflander) [inline src/vt-tv/render/render.cc:81]: Strange formatting here.

File: src/vt-tv/render/render.cc [hunk 4]
```diff
@@ -248,17 +265,11 @@ std::pair<double, double> Render::computeRankQoiRange_() {
...
-  if (qoi == "load"){
-    auto rank_loads_at_phase = this->info_.getAllRankLoadsAtPhase(phase);
-    for (auto [rank, rank_load] : rank_loads_at_phase){
-      rq_sum += rank_load;
-    }
-    return rq_sum / rank_loads_at_phase.size();
-  }
-  else{
-    throw std::runtime_error("Invalid QOI: " + qoi);
+  auto rank_loads_at_phase = this->info_.getAllRankQOIAtPhase(phase, qoi);
```
Comment (lifflander) [inline src/vt-tv/render/render.cc:268]: Does this need to be const& to avoid copies? Not sure of the signature

**Review by lifflander — CHANGES_REQUESTED** [review]

File: bindings/python/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+# # List of modules to bind
+# set(VT_TV_PYTHON_MODULES
+#   tv_from_json
+# )
+
+# Build the core parts of nanobind once
+nanobind_build_library(nanobind SHARED)
+
+# foreach(module ${VT_TV_PYTHON_MODULES})
```
[the remark this answers is not in the document]
Comment (cwschilly) [reply bindings/python/CMakeLists.txt]: Fixed

File: bindings/python/tv.h (hunk 1 above)
Comment (cwschilly) [reply bindings/python/tv.h]: Fixed

File: src/vt-tv/render/render.cc (hunk 3 above)
Comment (cwschilly) [reply src/vt-tv/render/render.cc]: Fixed

File: src/vt-tv/render/render.cc (hunk 4 above)
Comment (cwschilly) [reply src/vt-tv/render/render.cc]: Fixed

File: ci/build-and-test-bindings.dockerfile (hunk 2 above)
Comment (lifflander) [reply ci/build-and-test-bindings.dockerfile:1]: Let's ignore this issue for now and fix in in the follow-on issue for implementing full CI.

**Review by lifflander — APPROVED** [review]
Looks good to me

**Pull request merged**